src/cmd/palette/command.rs

```diff
@@ -26,13 +26,26 @@ pub struct PaletteImportArgs {
 
 /// Palette import command
 pub fn palette_import(g_args: &GlobalArgs, args: &PaletteImportArgs) -> Result<(), Error> {
-	let png_bytes = common::read_file_bin(&args.input_file)?;
-	let pal_config = VeraPaletteLoadConfig {
-		direct_load: true,
-		include_defaults: false,
-		sort: false,
+	let is_gpl = args.input_file.ends_with("gpl") || args.input_file.ends_with("GPL");
```

Comment:
Just wondering if there's a better way to do this rather than relying on file extension. I believe PNGs have magic bytes that can be used to identify them from raw data (first 8 bytes might be the same). A better workflow might be:

* Create an enum with `PNG` and `GPL` variants
* Load file as bytes
* Check first few bytes for png identifier or `Gimp Palette` (if that's consistent)
* Call a factored out `derive` function that takes the bytes and enum type and decides where to go from there

I might be getting ahead of myself a bit here, this is an easy enough refactor to do at a later date.

Reply:
Per my first comment, rolling a replacement for the `gimp_palette` crate is a first step toward enabling this sort of refactoring.
PNG files have an 8 byte header that could be used for identification (https://en.wikipedia.org/wiki/Portable_Network_Graphics#File_header).